test/models/AutomatonTest.java

```diff
@@ -13,113 +13,50 @@ public void testActionIsBothAnInputAndOutputThrowsException() {
         // Arrange
         Channel channel = new Channel("channel");
         Location location = Location.create("Location", new TrueGuard(), true, false, false, false);
-        List<Location> locations = new ArrayList<>();
-        locations.add(location);
         List<Edge> edges = new ArrayList<>();
-        edges.add(
-                new Edge(location, location, channel, true, new TrueGuard(), new ArrayList<>())
-        );
-        edges.add(
-                new Edge(location, location, channel, false, new TrueGuard(), new ArrayList<>())
-        );
-        List<Clock> clocks = new ArrayList<>();
-        List<BoolVar> booleans = new ArrayList<>();
+        edges.add(new Edge(location, location, channel, true, new TrueGuard(), new ArrayList<>()));
+        edges.add(new Edge(location, location, channel, false, new TrueGuard(), new ArrayList<>()));
 
         // Act
-        Automaton automaton = new Automaton(
-                "automaton",
-                locations,
-                edges,
-                clocks,
-                booleans,
-                false
-        );
+        new Automaton("automaton", location, edges);
     }
 
     @Test(expected = IllegalArgumentException.class)
-    public void testNoLocationsThrowsException() {
-        // Arrange
-        List<Location> locations = new ArrayList<>();
-        List<Edge> edges = new ArrayList<>();
-        List<Clock> clocks = new ArrayList<>();
-        List<BoolVar> booleans = new ArrayList<>();
-
+    public void testEmptyAutomatonThrowsIllegalArgumentException() {
         // Act
-        Automaton automaton = new Automaton(
-                "automaton",
-                locations,
-                edges,
-                clocks,
-                booleans,
-                false
-        );
+        new Automaton("automaton");
```

Comment:
Since you added this new `automaton` constructor, will it still throw an exception?

Reply:
Yes that is correct. It has been removed.